### PR TITLE
Search users

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Add search filtering to the user table. Fixes: https://github.com/canonical-web-and-design/MAAS-squad/issues/1368.

QA:
- Open the list of users.
- Type a username or email address into the searchbox.
- The user list should get filtered.

### CODE REVIEW

**Comment by squidsoup** [comment]
A wee bit of style weirdness here on the search cancel

image

**Comment by huwshimi** [comment]
@squidsoup that's just how the searchbox works: https://docs.vanillaframework.io/patterns/search-box/

**Comment by squidsoup** [comment]
> @squidsoup that's just how the searchbox works: https://docs.vanillaframework.io/patterns/search-box/

Hmm, you're right, but it looks broken don't you think?

**Comment by Caleb-Ellis** [comment]
There's an [issue on Vanilla](https://github.com/canonical-web-and-design/vanilla-framework/issues/1960) from Aug 2018 that I've just updated. We could have some simple overrides here (`u-no-margin--right` class and `background: transparent` on the reset button) in the meantime.

File: ui/src/app/settings/selectors/users.js [hunk 1]
```diff
@@ -13,6 +13,18 @@ users.get = (state, batch) => {
   return state.users.items;
 };
 
+/**
+ * Returns all users
+ * @param {Object} state - The redux state.
+ * @param {Number} batch - Number of users to return.
+ * @returns {Array} A list of all users.
+ */
```
Comment (Caleb-Ellis) [inline ui/src/app/settings/selectors/users.js]: Just need to update this doc block

Comment (huwshimi) [reply ui/src/app/settings/selectors/users.js]: Oops!

**Comment by huwshimi** [comment]
@Caleb-Ellis thanks, I've fixed up those buttons.

**Pull request merged**